### PR TITLE
Custom Cache Key for images

### PR DESCRIPTION
**Pull request by graphiclife**
Added parameter customCacheKey in af_setImage in UIImageView+AlamofireImage, UIButton+AlamofireImage.
Added parameter customCacheKey in ImageDownloader.download
Added test testThatImageCanBeCachedWithACustomCacheKey

### Goals :soccer:
Ability to cache images based on other factors than the url, which could be necessary if the backend hosting the image has a misconfigured cache management or provides images through REST-endpoints.

### Implementation Details :construction:
Added an optional cache key parameter to af_setImage-functions.

### Testing Details :mag:
Added testThatImageCanBeCachedWithACustomCacheKey in UIImageViewTests .

### CODE REVIEW

File: Source/ImageDownloader.swift [hunk 1]
```diff
@@ -263,6 +263,7 @@ open class ImageDownloader {
         _ urlRequest: URLRequestConvertible,
         receiptID: String = UUID().uuidString,
         filter: ImageFilter? = nil,
+        customCacheKey: String? = nil,
```
[the remark this answers is not in the document]
Comment (graphiclife) [reply Source/ImageDownloader.swift]: 👍

**Comment by graphiclife** [comment]
@cnoon Any updates on this?

**Comment by cnoon** [comment]
Not yet @graphiclife, right now we're very focused on the AF5 release and keeping AFI in sync with that. I'm not sure right now if this change will make it into AFI 4 or not.

**Comment by ricardolpd** [comment]
Hi guys, just wondering, do you have a timeline to get into master? I am having the same issue, and this approach would suffice. Is there anything I or someone could do to bring this to master?

**Comment by cnoon** [comment]
Hi @graphiclife, I have squashed these changes with a few edits and added tests in deff2079 and pushed the change into `master`. This will be released shortly as part of the 4.0.0 release. Thank you for all your hard work here and apologies for the poor turnaround time here.

Cheers. 🍻

**Comment by graphiclife** [comment]
@cnoon Excellent news, thanks!

**Pull request closed without merging**